packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -398,13 +421,35 @@ class _CupertinoSearchTextFieldState extends State<CupertinoSearchTextField>
     }
   }
 
+  void _onScroll() {
+    if (_ancestorScrollPosition == null || !_ancestorScrollPosition!.hasPixels) {
+      return;
+    }
+    final RenderBox? renderBox = context.findRenderObject() as RenderBox?;
+    final double currentHeight = renderBox?.size.height ?? 0.0;
+    setState(() { _fadeExtent = _calculateScrollOpacity(currentHeight, _maxHeight); });
+  }
+
+  static double _calculateScrollOpacity(double currentHeight, double maxHeight) {
+    final double thresholdHeight = maxHeight * _kMinHeightBeforeTotalTransparency;
```

Comment:
Question: should this only take height into account, or should padding be a factor? It might be fine to not get too granular in this to start though.

Reply:
It's a valid concern. Testing locally with different paddings, I don't think it will be a problem though. 

An additional benefit of making it dependent on the height alone is that it guarantees that by the time the search field is small enough (that the prefix/suffix icons overflow), the contents have faded out completely.